### DISCUSSION TITLE
HA alarm control panel custom bypass

### DISCUSSION THREAD

**Discussion by deanoddg** [rainepretorius/olarm-ha-integration#43] [Ideas]
Hi Raine

I have your Olarm integration running at my place though Home Assistant.
Its really great and I have to say a big thank you for this.

I have however not been able to use the custom bypass function on the HA control panel.
I do have an automation that allows me to bypass a zone and then arm the alarm, but i would like to be able to do this from the alarm control panel.
Is there something I am missing or is this not possible at the moment?

Thanks
Dean

**Comment by rainepretorius** [comment]
Hi @deanoddg 
It is not possible at the moment with the current config of Olarm's API.

**Reply by deanoddg** [reply]
Thanks Raine.

I ended up creating my own alarm card, kind of... just used the mushroom alarm card to show status and the tap actions as none. Then used a state-switch card with the various buttons corresponding to automations, in the automations I was able to bypass a zone using your bypass info and then delay 2 seconds followed by arming the alarm.
The state switch changes which buttons are displayed according to if the alarm is armed or not.
I'm still pretty new at all this, but thought I would find a way to get around my problem.

If anyone is interested here is the switching card code:
type: custom:state-switch
entity: template
template: >-
  {% if is_state('alarm_control_panel.main_house_house', 'armed_away') or
  is_state('alarm_control_panel.main_house_house','armed_home')  %}  armed 
  {%else %} disarmed {% endif %}
states:
  armed:
    type: custom:layout-card
    layout_type: custom:horizontal-layout
    layout:
      width: 100
      max_cols: 1
    cards:
      - type: custom:button-card
        name: DISARM
        styles:
          card:
            - height: 25px
          name:
            - color: '#03A9F4'
        tap_action:
          action: call-service
          service: automation.trigger
          service_data:
            entity_id: automation.disarm
  disarmed:
    square: false
    type: grid
    cards:
      - show_name: true
        show_icon: false
        type: custom:button-card
        name: ARM STAY
        styles:
          card:
            - height: 25px
            - padding: 2%
          name:
            - color: '#03A9F4'
        tap_action:
          action: call-service
          service: automation.trigger
          service_data:
            entity_id: automation.arm_stay
      - show_name: true
        show_icon: false
        type: custom:button-card
        name: ARM BYPASS
        styles:
          card:
            - height: 25px
            - padding: 2%
          name:
            - color: '#03A9F4'
        tap_action:
          action: call-service
          service: automation.trigger
          service_data:
            entity_id: automation.arm_bypass
      - show_name: true
        show_icon: false
        type: custom:button-card
        name: ARM ALL
        styles:
          card:
            - height: 25px
            - padding: 2%
          name:
            - color: '#03A9F4'
        tap_action:
          action: call-service
          service: automation.trigger
          service_data:
            entity_id: automation.arm_all

**Reply by rwoeke** [reply]
HI @deanoddg 

How did you get the Automation to work? I also have a zone that needs to be bypassed with every Arm, but an Automation could possibly do the trick as well. Do you mind sharing your automation?

**Reply by deanoddg** [reply]
Hi Rwoeke.

I created my own "alarm panel" which was 3 buttons and the mushroom alarm panel (which i just used for the status and set any button push to no action)
I then created 4 automations one for arm with bypass, one for arm stay, one for arm all and one for disarm.
From the buttons I used call service, automation trigger to start the automation.
In the automation I just used action Olarm arm away or Olarm arm Stay etc.
Most of the work is in the card not actually in the automations.
There is also a card state switch and this is to change the card so once you arm the alarm the card switches so you are able to disarm the alarm.
For the bypass automation i just used the Olarm bypass zone, and in data put zone_num: (and the zone you want to bypass),
I then gave a 2 second delay and after used Olarm Arm.

I hope all that makes sense. The code for the alarm card is above and the automations are pretty simple.
Let me know if you get it right.

**Reply by rwoeke** [reply]
Hi

Is the Format correct like this:


service: olarm_sensors.mon_bypass_zone
data: zone_num: 'BACK DOOR'

The zone is 12 and the alias is BACK DOOR

**Reply by deanoddg** [reply]
No just put 
data: zone_num: 12

**Reply by rwoeke** [reply]
thank you very much for helping @deanoddg 
Reference below for anyone else struggling with this.

```
alias: Olarm bypass
description: "Bypass Zone 12 whenever you arm"
trigger:
  - platform: state
    entity_id:
      - alarm_control_panel.indoor_mon
    to:
      - armed_away
      - armed_stay
      - armed_night
condition: []
action:
  - service: olarm_sensors.mon_bypass_zone
    data:
      zone_num: 12
mode: single
```
